oibackoff/index.d.ts

```diff
@@ -0,0 +1,46 @@
+// Type definitions for oibackoff
+// Project: https://github.com/chilts/oibackoff
+// Definitions by: Joshua DeVinney <https://github.com/geoffreak>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export declare function backoff(opts: BackoffOptions): BackoffInstance;
+
+export declare interface BackoffOptions {
+    maxTries?: number;
+    algorithm?: 'incremental' | 'exponential' | 'fibonacci';
+    delayRatio?: number;
+    maxDelay?: number;
+}
+
+export declare interface BackoffInstance {
+    (fn: Function, callback: Function): void;
```

Comment:
I was going to complain about the use of `Function` but it looks like this may be the best it can get? Have you tried using generic types and overloads for different arities?